CMakeLists.txt

```diff
@@ -1,17 +1,83 @@
 cmake_minimum_required(VERSION 3.10)
 
-project(cxxeffect)
+project(cxxeffect CXX)
 
-# add the executable
-add_executable(cxxeffect
-    src/stream.hpp
-    src/stream.cpp
-    src/task.hpp
-    src/task.cpp
-    src/eff.hpp
-    src/eff.cpp)
+# We want to export compile commands b/c it's useful for ccls
+set(CMAKE_EXPORT_COMPILE_COMMANDS true)
 
-set_property(TARGET cxxeffect PROPERTY CXX_STANDARD 17)
```

Comment:
Why was this removed? Isn't it helpful for IDEs and such?

Reply:
We could add it back in, although it'd have to be targeted towards C++20 since we'll be integrating coroutines!

Reply:
CCLS reads the build configuration from compile_commands.json so it doesn't need set_property, although if you know of other IDEs that rely on it, retaining it could be a good idea

Reply:
VSCode's CMake stuff relies on it by default. CLion also relies on it 😄